File: 387. First Unique Character in a String/solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        if not s:
            return -1

        res = len(s)
        for c in 'abcdefghijklmnopqrstuvwxyz':
            l, r = s.find(c), s.rfind(c)
            if l != -1 and l == r:
                res = min(res, l)

        return res if res != len(s) else -1

File: 387. First Unique Character in a String/solution.go
func firstUniqChar(s string) int {
	dic := [26]int{}
	for _, c := range s {
		dic[c-'a']++
	}
	for i, c := range s {
		if dic[c-'a'] == 1 {
			return i
		}
	}
	return -1
}